chapters/VK_QNX_external_memory_screen_buffer/device_memory.adoc

```diff
@@ -0,0 +1,125 @@
+pname:pNext chain of the slink:VkMemoryAllocateInfo structure.
+The sname:VkImportScreenBufferInfoQNX structure is defined as:
+
+include::{generated}/api/structs/VkImportScreenBufferInfoQNX.adoc[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:buffer is the QNX Screen buffer to import.
+
+The implementation may: not acquire a reference to the imported Screen
+buffer.
+Therefore, the application must: ensure that the object referred to by
+pname:buffer stays valid as long as the device memory to which it is
+imported is being used.
+
+.Valid Usage
+****
+  * If pname:buffer is not `NULL`, QNX Screen Buffers must: be supported for
+    import, as reported by slink:VkExternalImageFormatProperties or
+    slink:VkExternalBufferProperties
+  * If pname:buffer is not `NULL`, it must: be a valid QNX Screen buffer
+    object with code:SCREEN_PROPERTY_USAGE compatible with Vulkan.
+****
+
+include::{generated}/validity/structs/VkImportScreenBufferInfoQNX.adoc[]
+--
+
+[open,refpage='vkGetScreenBufferPropertiesQNX',desc='Get Properties of External Memory QNX Screen Buffers',type='protos']
+--
+To determine the memory parameters to use when importing a QNX Screen
+buffer, call:
+
+include::{generated}/api/protos/vkGetScreenBufferPropertiesQNX.adoc[]
+
+  * pname:device is the logical device that will be importing pname:buffer.
+  * pname:buffer is the QNX Screen buffer which will be imported.
+  * pname:pProperties is a pointer to a slink:VkScreenBufferPropertiesQNX
+    structure in which the properties of pname:buffer are returned.
+
+.Valid Usage
+****
+  * pname:buffer must: be a valid QNX Screen buffer object,
+    and must have code:_screen_buffer::code:SCREEN_PROPERTY_USAGE that includes
+    code:SCREEN_USAGE_VULKAN or code:SCREEN_USAGE_OPENGL_ES2 or
+    code:SCREEN_USAGE_OPENGL_ES3 or code:SCREEN_USAGE_NATIVE flags
+****
+
+include::{generated}/validity/protos/vkGetScreenBufferPropertiesQNX.adoc[]
+--
+
+[open,refpage='VkScreenBufferPropertiesQNX',desc='Properties of External Memory QNX Screen Buffers',type='structs']
+--
+The sname:VkScreenBufferPropertiesQNX structure returned is defined as:
+
+include::{generated}/api/structs/VkScreenBufferPropertiesQNX.adoc[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:allocationSize is the size of the external memory
+  * pname:memoryTypeBits is a bitmask containing one bit set for every
+    memory type which the specified Screen buffer can: be imported as.
+
+include::{generated}/validity/structs/VkScreenBufferPropertiesQNX.adoc[]
+--
+
+[open,refpage='VkScreenBufferFormatPropertiesQNX',desc='Structure describing the image format properties of a QNX Screen buffer',type='structs']
+--
+To obtain format properties of a QNX Screen buffer, include a
```

Comment:
Is it necessory to introduce new structure VkScreenBufferPropertiesQNX?
since it is not used by application, driver implementation should be able to query these properties from screen buffer object directly, I think we can remove it completely. 
vkGetScreenBufferPropertiesQNX can be removed as well, correct ?

Reply:
> Is it necessory to introduce new structure VkScreenBufferPropertiesQNX? since it is not used by application, driver implementation should be able to query these properties from screen buffer object directly, I think we can remove it completely. vkGetScreenBufferPropertiesQNX can be removed as well, correct ?

No, it cannot be removed. One of the examples is Angle, we don't want to use Screen API for screen_buffer_t inside the Angle, create format translation tables, etc. We will use this function instead to wrap EGL KHRImage without knowning details about the screen_buffer_t.